src/app/shell/personal-cabinet/provider/create-provider/create-provider.component.ts

```diff
@@ -152,7 +152,12 @@ export class CreateProviderComponent extends CreateFormComponent implements OnIn
     let res = [];     
     for (let el in form?.controls) {
       if (!form.get(el).value && form.get(el).status !== 'VALID') {
-        res.push(el)
+        res.push(el);
+      }
+      for (let elem in form?.get(el).value) {
+        if (form?.get(el).value[elem] === '') {
+          res.push(el);
+        }
```

Comment:
I still think you can simplify it even more
So, the button should be disabled if one of the required fields is not filled
then do via the form required controls and look for the FIRST invalid/empty control. You can combine the conditions in one row. Plus, you do not need to find all invalid cases, right? If there is at least one, then it is enough for disabling. Use find/some methods

Reply:
fixed, pls check)